brewtils/rest/system_client.py

```diff
@@ -519,19 +519,20 @@ def _wait_for_request(self, request, raise_on_error, timeout):
                 != self._easy_client.client.bg_host.upper()
                 or brewtils.plugin.CONFIG.bg_port != self._easy_client.client.bg_port
                 or brewtils.plugin.CONFIG.bg_url_prefix
-                != self._easy_client.client.bg_url_prefix
+                != self._easy_client.client.bg_prefix
             )
         ):
             request.parent = getattr(
                 brewtils.plugin.request_context, "current_request", None
             )
-            request.has_parent = True
-            ec = EasyClient(
-                bg_host=brewtils.plugin.CONFIG.bg_host,
-                bg_port=brewtils.plugin.CONFIG.bg_port,
-                bg_url_prefix=brewtils.plugin.CONFIG.bg_url_prefix,
-            )
-            ec.put_request(request)
+            if request.parent is not None:
+                request.has_parent = True
```

Comment:
I think we should upload the request regardless if there is a parent request. Because it could be created out of band from a traditional request and we still want to track that information. So we should do something like this:

```python
request.has_parent = request.parent is not None
```